### ISSUE TITLE
PHP Warning: array_replace_recursive(): Expected parameter 1 to be an array

### ISSUE THREAD

**Issue by Philscode1** [sudhaus7/typo3-xlsimport#24]
TYPO3 11.5.25
PHP 7.4

I got the error:
`PHP Warning: array_replace_recursive(): Expected parameter 1 to be an array, null given in /var/www/html/public/typo3/sysext/core/Classes/DataHandling/DataHandler.php line 1779`

**Comment by taunusweb** [comment]
The same with TYPO3 11.5.33 and PHP 8.0, any hints for that?

**Comment by foppelfb** [comment]
The Position is related to slug generation, but the variable at that point should not be null.. I wonder how that happens at all. can you provide an xls or csv that triggers this error?

**Comment by taunusweb** [comment]
Yes, this is my file (no personal Data, so it's fine for me to provide ist ;)

[Mappe1.xlsx](https://github.com/sudhaus7/typo3-xlsimport/files/13562007/Mappe1.xlsx)

**Comment by taunusweb** [comment]
I try to import the xlsx file into the tx_cartbooks_domain_model_book table, but I don't know, if I understand, what you mean...

**Comment by foppelfb** [comment]
Ok. I need more information. I tested this with a vanilla installation of this extension with EXT:cart_books, and I did not have any problems importing your dataset. 

xls-import uses the DataHandler to write the new records to the database, and of course it will run all registered hooks in that case, and apply configured actions from the TCA where applicable (like slug generation for example). 

1. Does your error message happen at the exact line as mentioned in the title?
2. Did you change or amend anything in the TCA configuration for the table tx_cartbooks_domain_model_book? 
3. are there other extensions present in the system that add to the DataHandler (in general or in relation to EXT:cart_books)?

**Comment by calien666** [comment]
@taunusweb @Philscode1 
I had the same issue today while reworking for v12 support.
Did you try to update records and has your user access for the slug field? The error encountered at mine, when the system tried to update the slug for an existing record.

**Comment by sbuerk** [comment]
> TYPO3 11.5.25 PHP 7.4
> 
> I got the error: `PHP Warning: array_replace_recursive(): Expected parameter 1 to be an array, null given in /var/www/html/public/typo3/sysext/core/Classes/DataHandling/DataHandler.php line 1779`

Just as a general advice - for these kind of errors/exceptions, it is always good to enable the debug mode and than provide the complete stack trace. This allows a extension maintainer/contributer to see the call chain and have some map at hand how it may come to this error in a TYPO3 core class.
